File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int n = s.size(), m = t.size();
        if(n==0) return true;
        if( n > m ) return false;
        int j = 0;
        for(int i = 0; i<m; i++){
            if(t[i] == s[j]) j++;
            if(j==n) return true;
        }
        return false;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    const [n,m] = [s.length,t.length];
    if(n>m) return false;
    let j = 0;
    for(let i = 0; i < m; i++){
        if(t[i] === s[j]) j++;
        if(j === n) return true;
    }
    return j === n;
};